fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/query/BaseWhereAdapter.java

```diff
@@ -0,0 +1,1137 @@
+        return getThis();
+    }
+
+    /**
+     * Add ESCAPE {str} to the expression
+     * @param str
+     * @return
+     */
+    public T escape(String str) {
+        predicateParser.escape();
+        predicateParser.literal(str);
+        return getThis();
+    }
+
+    /**
+     * Add ESCAPE {tableAlias}.{columnExpression} to the expression
+     * @param tableAlias
+     * @param columnExpression
+     * @return
+     */
+    public T escape(String tableAlias, String columnExpression) {
+        predicateParser.escape();
+        predicateParser.column(tableAlias, columnExpression);
+        return getThis();
+    }
+
+    /**
+     * Add ESCAPE ? to the expression
+     * @param bindMarker
+     * @return
+     */
+    public T escape(BindMarkerNode bindMarker) {
+        predicateParser.escape();
+        predicateParser.bindMarker(bindMarker);
+        return getThis();
+    }
+
+    /**
+     * Add the expression directly without any other operator
+     * @param filter
+     * @return
+     */
+    public T filter(ExpNode filter) {
+        predicateParser.addToken(filter);
+        return getThis();
+    }
+
+    /**
+     * Add a bind marker and its string value to the expression. Handles null.
+     * @param param
+     * @return
+     */
+    public T bind(String param) {
+        predicateParser.bindMarker(new StringBindMarkerNode(param));
+        return getThis();
+    }
+
+    /**
+     * Add a bind marker and Long value to the expression. Handles null.
+     * @param param
+     * @return
+     */
+    public T bind(Long param) {
+        predicateParser.bindMarker(new LongBindMarkerNode(param));
+        return getThis();
+    }
+
+    /**
+     * Add a bind marker and Integer value to the expression. Handles null.
+     * @param param
+     * @return
+     */
+    public T bind(Integer param) {
+        predicateParser.bindMarker(new IntegerBindMarkerNode(param));
+        return getThis();
+    }
+
+    /**
+     * Add a bind marker and Double value to the expression. Handles null.
+     * @param param
+     * @return
+     */
+    public T bind(Double param) {
+        predicateParser.bindMarker(new DoubleBindMarkerNode(param));
+        return getThis();
+    }
+
+    /**
+     * Add a bind marker and Instant value to the expression. Handles null.
+     * @param param
+     * @return
+     */
+    public T bind(Instant param) {
+        predicateParser.bindMarker(new InstantBindMarkerNode(param));
+        return getThis();
+    }
+
+    /**
+     * Add a bind marker and BigDecimal value to the expression. Handles null.
+     * @param param
+     * @return
+     */
+    public T bind(BigDecimal value) {
+        predicateParser.bindMarker(new BigDecimalBindMarkerNode(value));
+        return getThis();
+    }
+
+    /**
+     * Add the operator op to the expression
+     * @param op the enum of the operator to add
+     * @return
+     */
+    public T operator(Operator op) {
```

Comment:
what does this do exactly?

Reply:
It adds the requested operator to the expression. The calling code then needs to follow up with an operand to make a valid expression. This is used in some cases where the operator is a variable, not an explicit part of the code flow.

Reply:
got it! sorry I should have resolved this one... I got it in a number of classes later